### DISCUSSION TITLE
Uncaught Error: No matching key.

### DISCUSSION THREAD

**Discussion by kartikchoubisa** [WalletConnect/walletconnect-monorepo#2728] [Developer Support]
Hi, I am using the react wallet chat demo available [here](https://github.com/WalletConnect/web-examples/tree/main/wallets/react-wallet-chat).

```
_app-3a3aeafdf4ce7f98.js:1 Uncaught (in promise) Error: No matching key. history: 1687441832453713
    at ic.getRecord (_app-3a3aeafdf4ce7f98.js:1:2771673)
    at ic.get (_app-3a3aeafdf4ce7f98.js:1:2770296)
    at nc.onRelayEventResponse (_app-3a3aeafdf4ce7f98.js:1:2765592)
    at o.<anonymous> (_app-3a3aeafdf4ce7f98.js:1:2768494)
```

Everytime I send a message, I get multiple of these errors. 
I implemented chat on my own using the docs, I get the same error.

What are the consequences of this? I see that sometimes messages are lost, could this error be the reason?

Thanks!

Detailed log:
```
_app-3a3aeafdf4ce7f98.js:1 Uncaught (in promise) Error: No matching key. keychain: d046a7a0e253bcf2d55833288c57776610c2cb1631804916262289dc6b592044
    at Hr.get (_app-3a3aeafdf4ce7f98.js:1:2641789)
    at Kr.getSymKey (_app-3a3aeafdf4ce7f98.js:1:2644858)
    at Kr.decode (_app-3a3aeafdf4ce7f98.js:1:2643924)
    at o.<anonymous> (_app-3a3aeafdf4ce7f98.js:1:2673258)
    at o.emit (_app-3a3aeafdf4ce7f98.js:1:1305028)
    at fn.onMessageEvent (_app-3a3aeafdf4ce7f98.js:1:2662453)
    at async fn.onProviderPayload (_app-3a3aeafdf4ce7f98.js:1:2662312)
Hr.get @ _app-3a3aeafdf4ce7f98.js:1
getSymKey @ _app-3a3aeafdf4ce7f98.js:1
Kr.decode @ _app-3a3aeafdf4ce7f98.js:1
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onMessageEvent @ _app-3a3aeafdf4ce7f98.js:1
await in onMessageEvent (async)
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onMessageEvent @ _app-3a3aeafdf4ce7f98.js:1
await in onMessageEvent (async)
onProviderPayload @ _app-3a3aeafdf4ce7f98.js:1
await in onProviderPayload (async)
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onPayload @ _app-3a3aeafdf4ce7f98.js:1
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onPayload @ _app-3a3aeafdf4ce7f98.js:1
e.onmessage @ _app-3a3aeafdf4ce7f98.js:1
_app-3a3aeafdf4ce7f98.js:1 Uncaught (in promise) Error: No matching key. history: 1687442077122389
    at ic.getRecord (_app-3a3aeafdf4ce7f98.js:1:2771673)
    at ic.get (_app-3a3aeafdf4ce7f98.js:1:2770296)
    at nc.onRelayEventResponse (_app-3a3aeafdf4ce7f98.js:1:2765592)
    at o.<anonymous> (_app-3a3aeafdf4ce7f98.js:1:2768494)
getRecord @ _app-3a3aeafdf4ce7f98.js:1
get @ _app-3a3aeafdf4ce7f98.js:1
onRelayEventResponse @ _app-3a3aeafdf4ce7f98.js:1
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
await in (anonymous) (async)
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
await in (anonymous) (async)
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onMessageEvent @ _app-3a3aeafdf4ce7f98.js:1
await in onMessageEvent (async)
onProviderPayload @ _app-3a3aeafdf4ce7f98.js:1
await in onProviderPayload (async)
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onPayload @ _app-3a3aeafdf4ce7f98.js:1
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onPayload @ _app-3a3aeafdf4ce7f98.js:1
e.onmessage @ _app-3a3aeafdf4ce7f98.js:1
_app-3a3aeafdf4ce7f98.js:1 Uncaught (in promise) Error: No matching key. history: 1687442077122389
    at ic.getRecord (_app-3a3aeafdf4ce7f98.js:1:2771673)
    at ic.get (_app-3a3aeafdf4ce7f98.js:1:2770296)
    at Dp.onRelayEventResponse (_app-3a3aeafdf4ce7f98.js:1:2892789)
    at o.<anonymous> (_app-3a3aeafdf4ce7f98.js:1:2897913)
getRecord @ _app-3a3aeafdf4ce7f98.js:1
get @ _app-3a3aeafdf4ce7f98.js:1
Dp.onRelayEventResponse @ _app-3a3aeafdf4ce7f98.js:1
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
await in (anonymous) (async)
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
await in (anonymous) (async)
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onMessageEvent @ _app-3a3aeafdf4ce7f98.js:1
await in onMessageEvent (async)
onProviderPayload @ _app-3a3aeafdf4ce7f98.js:1
await in onProviderPayload (async)
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onPayload @ _app-3a3aeafdf4ce7f98.js:1
(anonymous) @ _app-3a3aeafdf4ce7f98.js:1
o.emit @ _app-3a3aeafdf4ce7f98.js:1
onPayload @ _app-3a3aeafdf4ce7f98.js:1
e.onmessage @ _app-3a3aeafdf4ce7f98.js:1

```

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/55321327/335c3516-3d50-4365-bb8a-322d7786e8d7)

**Comment by Barr-find** [comment]
Same ???

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/10328382/ae37e18d-e33f-4020-aede-273b519f58cd)

**Comment by Barr-find** [comment]
`localStorage.clear();` Helped!

**Reply by ramvi** [reply]
Well, it helped by turning off the restoration of the previous session. So by having the user make a new connection, the error goes away. I wouldn't call this a fix, but I guess it's a workaround.

**Comment by boidushya** [comment]
Hey, we are currently investigating this internally and you should see a fix very soon. But for the time being, as long as the property for No Matching Key is `history`, i.e, `No matching key. history: ...`, it shouldn't hamper the functionality of your dapp in any way. 
cc. @ganchoradkov

**Reply by ramvi** [reply]
Hi,

Thank you for the update, it's good to hear that a fix is on the way. 

I'd like to point out that even though the Dapp is still functional, I'm _unable to implement a try-catch block around the 'No Matching Key' history_. This results in several error messages being logged or directly shown to the users.

Hopefully, the upcoming fix will address this issue as well. Looking forward to it.

**Comment by damon-leong** [comment]
Hi,

I have this issue 
Unhandled Promise rejection: No matching key. keychain: 90c46da09ddeb42805e56de462e2baba6cc173e7a2e1fb6c0c7c02e847f4a1c7

**Comment by franfernandez20** [comment]
Hello, I'm also experiencing this issue. I will adopt the suggested solution and catch and auto-retry or make the user retry the connexion. But would be great a solution to it.

**Comment by AdrianKBL** [comment]
Good night,

I have been with this issue for too long , I have been trying too many different things but no one of these worked. We need a fix asap please!

**Comment by MickaelNeves** [comment]
Im using RainbowKit v^1.0.6 + WalletConnect v2 + Argent Wallet and I cant verify a signature
![Screenshot 2023-08-02 at 11 04 31](https://github.com/WalletConnect/walletconnect-monorepo/assets/20476271/f3bc6fb0-2184-4508-bc04-91f9c0b1836a)

**Comment by glitch-txs** [comment]
Aswered [here](https://github.com/orgs/WalletConnect/discussions/3291). Seems to be an issue on the Wallet side.

**Comment by dev-blc** [comment]
# Wallet Connect Issue: ``Error: No matching key``

## Description 
Error occured when one or many of Wallet connect requests are neither Accepted nor rejected, but the app is closed, quits or exited. 
This creates a pending request that produces the below error. 

`Uncaught (in promise) Error: No matching key. history: 1687441832453713`

## Solution 
Eventhough follwing all solutions listed all over the internet including clearing the local storage using both Inspect element's console and `localStorage.clear()` in JS code, the issue still persisted. 

We managed to solve the above error by the following steps 
- Go to the inspect element or window of your WebApp, dApp or Application. 
- Now navigate to `Storage` in the top navigation bar.
- Here you can find an Wallet Connect indexed DB under the dropdown `Indexed Databases`.
- Inside the folder, check if there is a table named `keyvaluestorage`.
- In this table you can see that the `"wc@2:client:0.3:request"` row is populated.
- Make sure this row is empty or delete the table using the bin logo in top right corner.
- Refresh your App/dApp window and try using wallect connect now.

#### NOTE
THE ABOVE SOLUTION IS A SUGGESTION THAT WORKED FOR US WITHOUT ANY COMPLICATIONS OR RAMIFICATIONS. PROCEED AT YOUR OWN RISK.
